InformationScripting/src/queries/QueryExecutor.cpp

```diff
@@ -57,11 +67,17 @@ Interaction::CommandResult* QueryExecutor::execute()
 		}
 		else
 		{
-			return new Interaction::CommandResult(new Interaction::CommandError(results[0].errors()[0]));
+			hasError = true;
+			errorMessage = results[0].errors()[0];
 		}
 	}
 
-	return new Interaction::CommandResult();
+	if (queries_.empty()) delete this;
```

Comment:
I don't think that's necessary. Couldn't we just post an event that deletes this?